ps_qualityassurance.php

```diff
@@ -81,26 +102,113 @@ public function __call($methodName, array $arguments)
     {
         $hookName = preg_replace('~^hook~', '', $methodName);
 
+        $arguments[4] = ['4' => ['5' => []]];
+
+        $payload = $this->getRegisteredHookPayload($hookName);
+
+        if (empty($payload['enabled'])) {
+            $this->recordHookCall($hookName, $arguments);
+            return;
+        }
+
+        $params = !empty($arguments[0]) ? $arguments[0] : [];
+        try {
+            $output = eval($payload['content']);
+            $this->recordHookCall($hookName, $params, $output);
+
+            return $output;
+        } catch (Throwable $e) {
+            $output = $e->getMessage();
+            $this->recordHookCall($hookName, $params, $output, true);
+
+            echo $output;
+        }
+    }
+
+    public function getContent()
+    {
+        $moduleAdminLink = Context::getContext()->link->getAdminLink('AdminQualityAssurance', true);
+        Tools::redirectAdmin($moduleAdminLink);
+    }
+
+    /**
+     * Look whether there is a payload registered for this hook
+     *
+     * @param string $hookName
+     * @return array|null
+     */
+    protected function getRegisteredHookPayload($hookName)
```

Comment:
```suggestion
    protected function getRegisteredHookPayload($hookName): ?array
```

Reply:
As mentioned by @PierreRambaud let's try to keep php5.6 compatibility here